lib/Gocdb_Services/NotificationService.php

```diff
@@ -136,12 +92,47 @@ public function roleRequest($entity) {
         if ($projectIds != null) {
             foreach ( $projectIds as $pid ) {
                 $project = \Factory::getOwnedEntityService ()->getOwnedEntityById ( $pid );
-                if(sendMail){
-                    $this->roleRequest ( $project );
-                } else {
-                    echo $project->getName () . "<br>";
-                }
+                $this->roleRequest ( $roleRequested, $requestingUser, $project );
+
             }
         }
     }
+
+
+    /**
+    * Return the PortalURL to enable an accurate link to the role approval view to be created
+    */
+    private function getWebPortalURL() {
+        return \Factory::getConfigService()->GetPortalURL();
+    }
+
+    private function sendEmail($roleRequested, $requestingUser, $entityName, $approvingUser) {
+        $subject = sprintf(
+            'GocDB: A Role request from %1$s over %2$s requires your attention',
```

Comment:
Not introduced by these changes, but I don't know why we would use a mixed case variation. 
```suggestion
            'GOCDB: A Role request from %1$s over %2$s requires your attention',
```